### PR TITLE
feat: not found page

### PR DESCRIPTION
**Pull request by DeniGolf**

### CODE REVIEW

**Review by aleksandr-kuharenko — APPROVED** [review]

File: frontend/src/components/main/main.tsx [hunk 1]
```diff
@@ -32,23 +33,18 @@ const Main: React.FC = () => {
...
+      <Route path={AppRoute.PAGE} component={withHeader(Pages)} exact />
+      <Route path={AppRoute.SETTINGS} component={withHeader(Settings)} />
+      <Route path={AppRoute.PROFILE} component={withHeader(ProfileInfo)} key={Date.now() />
+      <Route
+        path={AppRoute.WORKSPACE_SETTING}
+        component={withHeader(Workspace)}
+        exact
+      />
+      <Route path="/" component={withHeader(Pages)} exact />
+      <Route path="*" component={NotFound} />
```
Comment (mykytabatrak) [inline frontend/src/components/main/main.tsx:45]: The sole purpose of the `Main` component was to render things with `Toolbar` and `Header`, why is 404 page inside `Main` and not inside `App`?

File: frontend/src/components/not-found/not-found.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import { getAllowedClasses } from 'helpers/dom/dom';
+import { AppRoute } from 'common/enums/enums';
+import styles from './styles.module.scss';
+
+const NotFound: React.FC = () => {
+  return (
+    <Container
+      className={`${getAllowedClasses(
+        styles.notFoundWrapper,
+      )} d-flex flex-column`}
```
Comment (mykytabatrak) [inline frontend/src/components/not-found/not-found.tsx:12]: You don't need to wrap this in a template literal, `getAllowedClasses` accepts both strings and objects
```suggestion
className={getAllowedClasses(styles.notFoundWrapper), 'd-flex flex-column'}
```

File: frontend/src/components/not-found/not-found.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    >
+      <Row className="vh-100">
+        <Col sm={10} md={8} lg={6} className="mx-auto d-table vh-100">
+          <div className="d-table-cell align-middle">
```
Comment (mykytabatrak) [inline frontend/src/components/not-found/not-found.tsx:16]: Wouldn't flex-box be better to align something to the center
```suggestion
<Col sm={10} md={8} lg={6} className="d-flex flex-column justify-content-center vh-100">
```

**Review by mykytabatrak — CHANGES_REQUESTED** [review]

**Comment by AlikRakhmonov** [comment]
check if your build fails locally

**Review by mykytabatrak — APPROVED** [review]

**Pull request merged**